### PR TITLE
Added type annotation to navigator.py

### PR DESCRIPTION
**Pull request by anisstasyaadira**

### CODE REVIEW

File: morpcc/navigator.py [hunk 1]
```diff
@@ -65,14 +65,14 @@ def __init__(self, schema, entity, request):
...
     def add_attribute(
         self,
-        name,
-        type_,
-        title,
-        description=None,
-        required=False,
-        primary_key=False,
+        name: str,
+        type_: typing.Union[int, str, float],
```
Comment (kagesenshi) [inline morpcc/navigator.py:69]: u sure this accept int, str, float? .. read create_app.py .. you should see that appadd_attribute pass string there. 
```suggestion
        type_: str,
```

File: morpcc/navigator.py [hunk 2]
```diff
@@ -65,14 +65,14 @@ def __init__(self, schema, entity, request):
...
+        title: str,
+        description: str = None,
```
Comment (kagesenshi) [inline morpcc/navigator.py:71]: whenever a string variable defaults to None, the type should be typing.Optional[str]. Similarly on other fields.
```suggestion
        description: typing.Optional[str] = None,
```

File: morpcc/navigator.py [hunk 3]
```diff
@@ -344,7 +343,10 @@ def __init__(self, schema, request):
         self.entities = EntityBrowser(schema, request)
 
     def add_entity(
-        self, name, title, icon="database"
+        self, 
+        name: str, 
+        title: str, 
+        icon="database"
```
Comment (kagesenshi) [inline morpcc/navigator.py:349]: this should be a str type

File: morpcc/navigator.py [hunk 4]
```diff
@@ -410,14 +416,23 @@ def get_application(self, app_uuid):
...
         schema = self.schema_col.create(data, deserialize=False)
         if schema:
             return SchemaNavigator(schema, self.request)
 
     def add_application(
-        self, name, title, schema, icon="cube"
+        self,
+        name: str,
+        title: str,
+        schema,
```
Comment (kagesenshi) [inline morpcc/navigator.py:434]: whats the type annotation for this ?

File: morpcc/navigator.py [hunk 5]
```diff
@@ -65,14 +65,14 @@ def __init__(self, schema, entity, request):
...
-        description=None,
-        required=False,
-        primary_key=False,
+        name: str,
+        type_: str,
+        title: str,
+        description: typing.Optional[str] = None,
+        required: bool = False,
+        primary_key: bool = False,
         dictionaryelement=None,
```
Comment (kagesenshi) [inline morpcc/navigator.py:74]: where's the type annotation for this?

File: morpcc/navigator.py [hunk 6]
```diff
@@ -92,13 +92,13 @@ def add_attribute(
 
     def add_relationship(
         self,
-        name,
-        title,
+        name: str,
+        title: str,
         reference_attribute,
         reference_search_attribute,
```
Comment (kagesenshi) [inline morpcc/navigator.py:98]: missing type annotation

File: morpcc/navigator.py [hunk 7]
```diff
@@ -115,13 +115,12 @@ def add_relationship(
 
     def add_backrelationship(
         self,
-        name,
-        title,
+        name: str,
+        title: str,
         reference_relationship,
```
Comment (kagesenshi) [inline morpcc/navigator.py:120]: missing type annotation

File: morpcc/navigator.py (hunk 6 above)
Comment (anisstasyaadira) [reply morpcc/navigator.py:98]: How do I know this type?
In create_app.py: 
reference_attribute=cifid.attributes["cifid"],
reference_search_attribute=cifid.attributes["cifid"]

Comment (kagesenshi) [reply morpcc/navigator.py:98]: from studying the code, what type you think output of cifid.attributes['cifid']?

Comment (kagesenshi) [reply morpcc/navigator.py:98]: you can also try including breakpoint() before that function call in create_app.py .. and try to run the script using morpfw execute .. that'll allow you to inspect the contents

File: morpcc/navigator.py [hunk 8]
```diff
@@ -65,14 +65,14 @@ def __init__(self, schema, entity, request):
...
-        primary_key=False,
-        dictionaryelement=None,
-        allow_invalid=False,
+        name: str,
+        type_: str,
+        title: str,
+        description: typing.Optional[str] = None,
+        required: bool = False,
+        primary_key: bool = False,
+        dictionaryelement: typing.Optional[str] = None,
```
Comment (kagesenshi) [inline morpcc/navigator.py:74]: nope .. 

read ``create_app.py`` and ``navigator.py``
-  ``nav`` variable stores ``Navigator`` 
- ``dd`` variable get from ``nav.datadictionary``, which if your read ``navigator.py``, ``datadictionary`` attribute of ``Navigator`` is a ``DataDictionaryBrowser`` object.
- ``dd['cifid']`` will call ``DataDictionaryBrowser.__getitem__('cifid')``, which returns ``DictionaryEntityNavigator``

so whats type type of ``dictionaryelement`` ?

File: morpcc/navigator.py [hunk 9]
```diff
@@ -92,13 +92,13 @@ def add_attribute(
...
-        title,
-        reference_attribute,
-        reference_search_attribute,
-        description=None,
-        required=False,
-        primary_key=False,
+        name: str,
+        title: str,
+        reference_attribute: typing.Optional[str] = None,
+        reference_search_attribute: typing.Optional[str] = None,
```
Comment (kagesenshi) [inline morpcc/navigator.py:98]: read comment on ``dictionaryelement``

File: morpcc/navigator.py [hunk 10]
```diff
@@ -115,13 +115,12 @@ def add_relationship(
...
     def add_backrelationship(
         self,
-        name,
-        title,
-        reference_relationship,
-        description=None,
-        single_relation=False,
+        name: str,
+        title: str,
+        reference_relationship: typing.Optional[str] = None,
```
Comment (kagesenshi) [inline morpcc/navigator.py:120]: check comment on ``dictionaryelement``

**Pull request merged**